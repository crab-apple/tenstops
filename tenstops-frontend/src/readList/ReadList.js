import React from 'react'
import PropTypes from 'prop-types'
import { Loader } from 'semantic-ui-react'
import ReadListItems from './ReadListItems'

export default function ReadList ({reads}) {
  if (reads) {
    return <ReadListItems reads={reads} />
  } else {
    return <Loader active />
  }
}

ReadList.propTypes = {
  reads: PropTypes.arrayOf(PropTypes.object)
}
