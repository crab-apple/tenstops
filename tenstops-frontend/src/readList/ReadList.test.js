/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import ReadList from './ReadList'
import { MemoryRouter } from 'react-router-dom'

test('snapshot', () => {
  const reads = [
    {
      id: 1,
      title: 'Novel 1',
      author: 'John Doe 1'
    }, {
      id: 2,
      title: 'Novel 2',
      author: 'John Doe 2'
    }, {
      id: 3,
      title: 'Novel 3',
      author: 'John Doe 3'
    }
  ]

  const component = renderer.create(
    <MemoryRouter>
      <ReadList reads={reads} />
    </MemoryRouter>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
