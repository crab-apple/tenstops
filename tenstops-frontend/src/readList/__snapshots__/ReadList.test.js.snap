// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui divided relaxed list"
  role="list"
>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <a
      href="/read/1"
      onClick={[Function]}
    >
      <div
        className="header"
      >
        Novel 1
      </div>
      John Doe 1
    </a>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <a
      href="/read/2"
      onClick={[Function]}
    >
      <div
        className="header"
      >
        Novel 2
      </div>
      John Doe 2
    </a>
  </div>
  <div
    className="item"
    data-value={undefined}
    onClick={[Function]}
    role="listitem"
  >
    <a
      href="/read/3"
      onClick={[Function]}
    >
      <div
        className="header"
      >
        Novel 3
      </div>
      John Doe 3
    </a>
  </div>
</div>
`;
